okhttp/src/jvmMain/kotlin/okhttp3/internal/-UtilJvm.kt

```diff
@@ -223,12 +223,10 @@ fun <T> immutableListOf(vararg elements: T): List<T> {
   return Collections.unmodifiableList(listOf(*elements.clone()))
 }
 
-/** Returns an immutable copy of this. */
-internal fun <K, V> Map<K, V>.toImmutableMap(): Map<K, V> {
-  return if (isEmpty()) {
-    emptyMap()
-  } else {
-    Collections.unmodifiableMap(LinkedHashMap(this))
+internal actual fun <K, V> Map<K, V>.toImmutableMap(): Map<K, V> {
+  return when {
+    isEmpty() -> emptyMap()
+    else -> Collections.unmodifiableMap(LinkedHashMap(this))
```

Comment:
Is this because they could have grabbed the tags from the builder and downcasted it? How do you decide to protect against this case?

Reply:
now that you mention it, this is probably overkill. It's to prevent Java callers mutating the map, but I don't think we expose the map directly anyway. Can simplify.